core/src/main/java/com/redhat/lightblue/client/Locking.java

```diff
@@ -49,4 +52,40 @@ public int getLockCount(String resourceId) throws LightblueException {
     public boolean ping(String resourceId) throws LightblueException {
         return ping(callerId,resourceId);
     }
+
+    public Lock lock(String resourceId) throws LightblueException {
+        return lock(resourceId, null);
+    }
+
+    public Lock lock(String resourceId, Long ttl) throws LightblueException {
+        acquire(resourceId, ttl);
+        return new Lock(resourceId);
+    }
+
+    public class Lock implements Closeable {
+
+        private final String resourceId;
+
+        public Lock(String resourceId) {
+            this.resourceId = resourceId;
+        }
+
+        public boolean ping() throws LightblueException {
+            return Locking.this.ping(resourceId);
+        }
+
+        public int getLockCount() throws LightblueException {
+            return Locking.this.getLockCount(resourceId);
+        }
+
+        @Override
+        public void close() throws IOException {
+            try {
+                release(resourceId);
+            } catch (LightblueException e) {
+                throw new IOException("Unable to release lock: " + resourceId, e);
```

Comment:
I am still debating about this re-throw as an IOException. Everywhere else the LB exception bubble up, so this is a notable deviation from that. Additionally, I don't think of the LB exception as a IOException. The alternative is to use a RuntimeException, but that doesn't solve the first issue and is now unchecked.